core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -91,6 +97,63 @@ public DeterministicKey(ImmutableList<ChildNumber> childNumberPath,
         this.keyCrypter = checkNotNull(crypter);
     }
 
+    /**
+     * Return the fingerprint of this key's parent as an int value, or zero if this key is the
+     * root node of the key hierarchy.  Raise an exception if the arguments are inconsistent.
+     * This method exists to avoid code repetition in the constructors.
+     */
+    private int ascertainParentFingerprint(DeterministicKey parentKey, int parentFingerprint)
+    throws IllegalArgumentException {
+        if (parentFingerprint != 0) {
+            if (parent != null)
+                checkArgument(parent.getFingerprint() == parentFingerprint,
+                              "parent fingerprint mismatch",
```

Comment:
Doesn't this need %s %s in it to print the arguments? And if so wouldn't 0x%x 0x%x work too?